### PR TITLE
merge CAE finetune

### PR DESCRIPTION
**Pull request by zengshao0622**

### CODE REVIEW

File: ppcls/configs/CAE/cae_base_patch16_224.yaml [hunk 1]
```diff
@@ -0,0 +1,170 @@
+# global configs
```
Comment (weisy11) [inline ppcls/configs/CAE/cae_base_patch16_224.yaml:1]: 文件名注明一下是finetune配置，如cae_blabla_finetune.yaml

File: ppcls/arch/backbone/model_zoo/cae.py [hunk 2]
```diff
@@ -0,0 +1,795 @@
...
+    def parse(x):
+        if isinstance(x, collections.abc.Iterable):
+            return x
+        return tuple(repeat(x, n))
+    return parse
+
+to_2tuple = _ntuple(2)
+
+zeros_ = nn.initializer.Constant(value=0.)
+ones_ = nn.initializer.Constant(value=1.)
```
Comment (weisy11) [inline ppcls/arch/backbone/model_zoo/cae.py:24]: 这个是从cae的代码copy过来的吗？看起来不太规范啊。尽量避免用全局变量吧。另外这里的zeros和ones是用来初始化吗？可以看看用其他的方法来做。

File: ppcls/arch/backbone/model_zoo/cae.py [hunk 3]
```diff
@@ -0,0 +1,795 @@
+import collections
```
Comment (weisy11) [inline ppcls/arch/backbone/model_zoo/cae.py:1]: 检查下与https://github.com/PaddlePaddle/VIMER/tree/main/CAE 里的网络结构代码是否一致，一致的话这里加个引用说明注释。

File: ppcls/arch/backbone/model_zoo/cae.py [hunk 4]
```diff
@@ -0,0 +1,795 @@
...
+                    f = interpolate.interp2d(x, y, z, kind='cubic')
+                    all_rel_pos_bias.append(
+                        paddle.Tensor(f(dx, dy)).contiguous().view(-1, 1).to(rel_pos_bias.device))
+
+                rel_pos_bias = paddle.concat(all_rel_pos_bias, axis=-1)
+
+                new_rel_pos_bias = paddle.concat((rel_pos_bias, extra_tokens), axis=0)
+                checkpoint_model[key] = new_rel_pos_bias
+
+    print("##############new keys:", len(checkpoint_model), checkpoint_model.keys())
```
Comment (weisy11) [inline ppcls/arch/backbone/model_zoo/cae.py:595]: 检查下代码里的print，不必要的全都去掉，必要的用logger

File: ppcls/arch/backbone/model_zoo/cae.py [hunk 5]
```diff
@@ -0,0 +1,795 @@
...
+import numpy as np
+from functools import partial
+
+import paddle
+import paddle.nn as nn
+import paddle.nn.functional as F
+
+import copy
+import os
+
```
Comment (weisy11) [inline ppcls/arch/backbone/model_zoo/cae.py:13]: 以上引用都是有用的吗？用不到的都去掉吧。

File: ppcls/data/preprocess/batch_ops/batch_operators.py [hunk 6]
```diff
@@ -229,3 +231,232 @@ def __call__(self, batch):
             list(self.ops.keys()), weights=list(self.ops.values()), k=1)[0]
         # return batch directly when None Op
         return op(batch) if op else batch
+
+
+class HackedMixupOperator(object):
```
Comment (weisy11) [inline ppcls/data/preprocess/batch_ops/batch_operators.py:236]: 这个类能不能换个名字？比如可以体现一下类的作用或优化点

File: ppcls/engine/engine.py [hunk 7]
```diff
@@ -501,7 +501,7 @@ def export(self):
         assert self.mode == "export"
         use_multilabel = self.config["Global"].get(
             "use_multilabel",
-            False) and "ATTRMetric" in self.config["Metric"]["Eval"][0]
+            False) or "ATTRMetric" in self.config["Metric"]["Eval"][0]
```
Comment (weisy11) [inline ppcls/engine/engine.py:504]: 这里为啥改成or

File: ppcls/loss/softtargetceloss.py [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+    def __init__(self):
+        super().__init__()
+
+    def forward(self, x, target):
+        loss = paddle.sum(-target * F.log_softmax(x, axis=-1), axis=-1)
+        loss = loss.mean()
+        return {"SoftTargetCELoss": loss}
+    
+    def __str__(self,):
+        return type(self).__name__
```
Comment (weisy11) [inline ppcls/loss/softtargetceloss.py:15]: 这个Pr过了precommit了吗？为啥git报格式warning

File: ppcls/optimizer/optimizer.py [hunk 9]
```diff
@@ -272,3 +272,144 @@ def __call__(self, model_list):
...
+            if 'teacher' not in param.name
+            ]
+            weight_decay = 0.
+        else:
+            parameters = model.parameters()
+
+        opt_args = dict(learning_rate=self.learning_rate, weight_decay=self.weight_decay)
+        opt_args['parameters'] = parameters
+        if decay_dict is not None:
+            opt_args['apply_decay_param_fun'] = lambda n: decay_dict[n]
```
Comment (weisy11) [inline ppcls/optimizer/optimizer.py:399]: 减少使用比较高端的语法，有的开发者会看不懂…

File: ppcls/optimizer/optimizer.py [hunk 10]
```diff
@@ -272,3 +272,144 @@ def __call__(self, model_list):
...
+                name_dict=None,
+                name=None):
+            if not isinstance(layerwise_decay, float) and \
+                    not isinstance(layerwise_decay, fluid.framework.Variable):
+                raise TypeError("coeff should be float or Tensor.")
+            self.layerwise_decay = layerwise_decay
+            self.n_layers = n_layers
+            set_param_lr_fun = self._layerwise_lr_decay
+            self.set_param_lr_fun = partial(set_param_lr_fun, layerwise_decay,
+                                            name_dict, n_layers)
```
Comment (weisy11) [inline ppcls/optimizer/optimizer.py:326]: 以上几行代码看起来很迷…
